list-objects-filter.c

```diff
@@ -620,6 +621,98 @@ static void filter_object_type__init(
 	filter->free_fn = free;
 }
 
+/*
+ * A filter which passes the objects to a compile-time plugin.
+ * The plugin needs to implement the filter_profile_plugin interface
+ * defined in list-objects-filter.h.
+ */
+
+#ifndef FILTER_PROFILE_PLUGINS
+#define FILTER_PROFILE_PLUGINS
+#endif
+
+static const struct filter_profile_plugin
+	*filter_profile_plugins[] = {
+		FILTER_PROFILE_PLUGINS
+	};
+
+struct filter_profile_data {
+	const struct filter_profile_plugin *plugin;
+	void *context;
+};
+
+static enum list_objects_filter_result filter_profile_filter_object(
+	struct repository *r,
+	enum list_objects_filter_situation filter_situation,
+	struct object *obj,
+	const char *pathname,
+	const char *filename,
+	struct oidset *omits,
+	void *filter_data_)
+{
+	struct filter_profile_data *filter_data = filter_data_;
+
+	enum list_objects_filter_omit omit_it = LOFO_IGNORE;
+
+	enum list_objects_filter_result ret =
+		filter_data->plugin->filter_object_fn(
+			r,
+			filter_situation,
+			obj,
+			pathname,
+			filename,
+			&omit_it,
+			&filter_data->context);
+
+	if (omits) {
+		if (omit_it == LOFO_KEEP)
+			oidset_remove(omits, &obj->oid);
+		else if (omit_it == LOFO_OMIT)
+			oidset_insert(omits, &obj->oid);
+	}
+	return ret;
+}
```

Comment:
I suppose so? Since you actually wrote the filter - you designed the profile filter interface, and adapted the builtin interface to the profile interface - you will have a better idea than me if there is anything that is missing from the second interface.